renku/core/utils/urls.py

```diff
@@ -110,3 +111,10 @@ def get_slug(name):
     valid_end = re.sub(r"[._-]$", "", valid_start)
     no_dot_lock_at_end = re.sub(r"\.lock$", "_lock", valid_end)
     return no_dot_lock_at_end
+
+
+def validate_url(repo: str) -> ParseResult:
+    """Validates the supplied url and returns the parsed URL if valid."""
+    if re.search(_URL_VALIDATOR, repo, re.ASCII):
+        raise errors.ParameterError(f"Invalid url: `{repo}`")
+    return urllib.parse.urlparse(repo)
```

Comment:
We use url.netloc in a couple locations, but for instance `"not-a-url"` would pass through this function and return a netloc of `""`. Maybe we should check that there is a netloc and if not raise a renku.ParameterError as well?

The exception would be with `file:///some/path/to/a/file` which also doesn't have a netloc but could potentially be valid. But current, we don't support these kinds of urls anyways.